zookeeper/init.sls

```diff
@@ -33,15 +33,8 @@ install-zookeeper:
     - group: root
     
 zookeeper-home-link:
-  alternatives.install:
-    - link: {{ zk.alt_home }}
-    - path: {{ zk.real_home }}
-    - onlyif: test -d {{ zk.real_home }} && test ! -L {{ zk.alt_home }}
-    - priority: 30
-    - require:
-      - archive: install-zookeeper
   file.symlink:
     - name: {{ zk.alt_home }}
     - target: {{ zk.real_home }}
```

Comment:
Actually no. `alternatives.install` doesn't work in this way. `alternatives.install` installs the alternative (which also creates a symlink) only in the situation when there is no already installed alternative. In case when there is the alternative with a different path and with the same prioriy SaltStack will show the error. Because it's not possible to setup same link with the same priority. 
What does it mean? Just try to remove `file.symlink`, then install zookeeper 3.4.9 on a testing cluster and after the successfulinstallation, try to update your zookeeper to the new version 3.4.10. You will get the error and your zookeeper 3.4.10 won't be installed because the alternative  already exists with the different path but with the same priority. That is why I don't understand why we are using alternatives in this formula. 
Logically, only one alternative is possible that is why you won't be able to change your zookeeper version from 3.4.9 to 3.4.10 or vice versa via alternatives. 
`file.symlink` fixes it. And now it's possible to install newer version of software but still you always will have only one alternative (which is kind of buggy and not useful). I will try all this stuf tomorrow, maybe I missed something